### PR TITLE
Add warning to widgets when column is missing

### PR DESCRIPTION
**Pull request by sclebal**
image

- Implemented InvalidColumnError to throw it when the column used is invalid.
- Used InvalidColumnError to throw errors in `stats.js` and `features.worker.js`.
- Adapted HistogramWidget minMax fetching and useWidgetFetch to handle InvalidColumnError.

### CODE REVIEW

**Review by zbigg — CHANGES_REQUESTED** [review]

File: packages/react-core/src/utils/InvalidColumnError.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const NAME = 'InvalidColumnError';
+const MESSAGE_START = `Uncaught ${NAME}:`;
+
+export class InvalidColumnError extends Error {
+  static message =
+    'The column selected for this widget is no longer available in the data source.';
+
+  constructor(message) {
```
Comment (zbigg) [inline packages/react-core/src/utils/InvalidColumnError.js:8]: My proposition would be to change this to

super(`ERR_INVALID_COLUMN: ${message || default}`)

and in is just test for existence of `ERR_INVALID_COLUMN: ` in error message.

This is still _string_ test, but at least based on something that looks like error code and _is_ universal and doesn't depend on particular details like adding "Uncaught..." by particular runtime.

File: packages/react-widgets/__tests__/hooks/useWidgetFetch.test.js [hunk 2]
```diff
@@ -79,17 +80,45 @@ describe('useWidgetFetch', () => {
     await act(() => sleep(250));
     expect(screen.queryByText('loading')).not.toBeInTheDocument();
 
-    expect(onError).toBeCalled();
+    expect(onError).toBeCalledTimes(1);
+
+    modelFn.mockRejectedValue(
+      new InvalidColumnError('Uncaught InvalidColumnError: Invalid column')
+    );
```
[the remark this answers is not in the document]
Comment (Clebal) [reply packages/react-widgets/__tests__/hooks/useWidgetFetch.test.js:87]: Idk why when I throw the error in the test the `Uncaught InvalidColumnError` label doesn't appear but this is the classic label `Uncaught {error class name}` that appears always in JS, that's why I used it to detect the error in the `is` fn.

File: packages/react-core/src/utils/InvalidColumnError.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const NAME = 'InvalidColumnError';
+const MESSAGE_START = `Uncaught ${NAME}:`;
+
+export class InvalidColumnError extends Error {
+  static message =
+    'The column selected for this widget is no longer available in the data source.';
```
Comment (zbigg) [inline packages/react-core/src/utils/InvalidColumnError.js:6]: Minor: i don't think message copy belongs to "react-core" and exception type, it's purely widgets & UI matter (and it's used only once there) so maybe move it there?

This should be in `[useWidgetFetch.js]` (or in UI component).

Comment (Clebal) [reply packages/react-core/src/utils/InvalidColumnError.js:6]: It used in @carto/react-api also.

Comment (Clebal) [reply packages/react-core/src/utils/InvalidColumnError.js:6]: Oh, now I read it correctly. I can move the message to the useWidgetFetch, ok.

**Review by zbigg — APPROVED** [review]
LGTM :)

**Pull request merged**